Marshal/MarshaledObject.swift

```diff
@@ -14,25 +14,33 @@
 import Foundation
 
 
-// MARK: - Types
-
-public typealias MarshaledObject = [String: AnyObject]
-
-
-// MARK: - Dictionary Extensions
+public protocol MarshaledObject {
+    subscript(key: KeyType) -> Any? { get }
+    func anyForKey(key: KeyType) throws -> Any
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A?
+    func valueForKey<A: ValueType>(key: KeyType) throws -> [A]
+    func valueForKey<A: ValueType>(key: KeyType) throws -> [A]?
+    func valueForKey<A: ValueType>(key: KeyType) throws -> Set<A>
+    func valueForKey<A: ValueType>(key: KeyType) throws -> Set<A>?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> A?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> [A]
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> [A]?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> Set<A>
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> Set<A>?
+}
 
```

Comment:
I have already adjusted the current protocol to only have the `subscript` and `anyForKey`. I like the idea of having `anyForKey` because people have mentioned that they would like to change the separator. However, I am not against removing it from the protocol if that is what everyone wants.